File: particle-connect/android/src/main/kotlin/network/particle/flutter/bridge/utils/ChainUtils.java
package network.particle.flutter.bridge.utils;

import com.particle.base.ChainInfo;
import com.particle.base.ChainName;

import java.lang.reflect.Constructor;


public class ChainUtils {

    public static ChainInfo getChainInfo(String chainName, String chainIdName) {
        if (ChainName.BSC.toString().equals(chainName)) {
            chainName = "Bsc";
        }
        try {
            Class clazz1 = Class.forName("com.particle.base." + chainName + "Chain");
            Constructor cons = clazz1.getConstructor(String.class);
            ChainInfo chainInfo = (ChainInfo) cons.newInstance(chainIdName);
            return chainInfo;
        } catch (Exception e) {
            e.printStackTrace();
            throw new RuntimeException(e.getMessage());
        }
    }
}

File: particle-biconomy/android/src/main/kotlin/network/particle/biconomy_flutter/bridge/utils/ChainUtils.kt
package network.particle.biconomy_flutter.bridge.utils

import com.particle.base.ChainInfo
import com.particle.base.ChainName

object ChainUtils {
    fun getChainInfo(chainName: String, chainIdName: String?): ChainInfo {
        var chainName = chainName
        if (ChainName.BSC.toString() == chainName) {
            chainName = "Bsc"
        }
        return try {
            val clazz1 = Class.forName("com.particle.base." + chainName + "Chain")
            val cons = clazz1.getConstructor(String::class.java)
            cons.newInstance(chainIdName) as ChainInfo
        } catch (e: Exception) {
            e.printStackTrace()
            throw RuntimeException(e.message)
        }
    }
}